include/SharedMemory.h

```diff
@@ -71,6 +75,26 @@ class SharedMemoryData
 	void* m_ptr = nullptr;
 };
 
+	static inline std::string createKey() {
+#if defined(LMMS_BUILD_APPLE) || defined(LMMS_BUILD_FREEBSD) || defined(LMMS_BUILD_OPENBSD)
+		const int &length = 30; // PSHMNAMLEN=31
+#else
+		const int &length = 32; // mimic UUID
+#endif
+		std::string key;
+		std::random_device rd;
+		std::mt19937 gen(rd()); // mersenne twister, seeded
+		std::uniform_int_distribution<> distrib(0, 15); // hex range (0-15)
+
+		for (int i = 0; i < length; i++) {
+			std::stringstream ss;
+			ss << std::hex << distrib(gen);
+			key += ss.str();
+		}
```

Comment:
```suggestion
		key.reserve(length);	
		for (int i = 0; i < length; i++) {
			key += "0123456789ABCDEF"[distrib(gen)];
		}
```
This is more efficient and will let you remove `<sstream>`.